File: elixer/leetcode_elixer/lib/leetcode_elixir/fizz_buzz412.ex
defmodule LeetcodeElixir.FizzBuzz412 do
  @moduledoc """
  This module has one function: `calculate_fizz_buzz/1`.

  `calculate_fizz_buzz/1` loops through numbers up to a given limit,
  appending strings to the output list based on divisibility:

  - If a number is divisible by both 3 and 5, append 'FizzBuzz'.
  - If a number is divisible by 5, append 'Buzz'.
  - If a number is divisible by 3, append 'Fizz'.

  ## Examples

      iex> FizzBuzz412.calculate_fizz_buzz(16)
      ["Fizz", "Buzz", "Fizz", "Fizz", "Buzz", "Fizz", "FizzBuzz", ...]
  """
  import Utils

  # map takes and returns a list
  # map each item in the list to a function
  # 1..limit is input list
  def calculate_fizz_buzz(limit) do
    puts "limit: #{limit}"
    1..limit
    |> Enum.map(fn
      n when rem(n, 15) == 0 -> "FizzBuzz"
      n when rem(n, 5) == 0 -> "Buzz"
      n when rem(n, 3) == 0 -> "Fizz"
      n -> n
    end)
  end
end

File: python/leetcode/fizz_buzz_412.py
def calculate_fizz_buzz():
    output = []

    for n in range(1, 16):
        if n % 3 == 0 and n % 5 == 0:
            output.append('FizzBuzz')
        elif n % 5 == 0:
            output.append('Buzz')
        elif n % 3 == 0:
            output.append('Fizz')
        else:
            output.append(str(n))

    print(output)

    output = ["FizzBuzz" if n % 3 == 0 and n % 5 == 0
              else "Fizz" if n % 3 == 0
              else "Buzz" if n % 5 == 0
              else str(n)
              for n in range(1, 16)]